src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -5597,6 +5597,65 @@ static void H()
                 );
         }
 
+        [Fact]
+        public void GeneratedSyntaxTrees_Nullable()
+        {
+            var source1 = CSharpSyntaxTree.ParseText(@"
+partial class Program
+{
+    static void G(object x)
+    {
+        x = null; // no warning
+        F = null; // no warning
+    }
+}", isGeneratedCode: true);
+
+            Assert.True(((CSharpSyntaxTree)source1).IsGeneratedCode());
+
+            var source2 = CSharpSyntaxTree.ParseText(@"
+partial class Program
+{
+    static object F = new object();
+
+    static void H(object x)
+    {
+        x = null; // warning 1
+        F = null; // warning 2
+    }
+}", isGeneratedCode: false);
+
+            Assert.False(((CSharpSyntaxTree)source2).IsGeneratedCode());
+
+            var source3 = CSharpSyntaxTree.ParseText(
+@"
+partial class Program
+{
+    static void I(object x)
+    {
+        x = null; // warning 3
+        F = null; // warning 4
+    }
+}", isGeneratedCode: null /* use heuristic */);
```

Comment:
>/* use heuristic */ [](start = 26, length = 19)

I'm not really sure what this comment is supposed to indicate. Consider expanding or removing.